### PR TITLE
Adding docker compose file for both development and production

### PR DESCRIPTION
**Pull request by D-sourav-M044**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #313 

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [X] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Docker compose file for development and production has been added.
## What was done
- Static files were built and copied to the nginx server. Created two different docker-compose for development and production.
## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

File: Todo-List-App/Dockerfile.prod [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#Using node as required run time to get base image.
+#Alpine images are smaller, requires less memory and gives better performance
+FROM node:18.7.0-alpine as builder
+
+#defining a working directory in container. Commands will be executed from this directory.
+WORKDIR /app
+
+#copying package.json file to install the required dependency to run the project.
+COPY package.json .
```
Comment (musfique-m) [inline Todo-List-App/Dockerfile.prod:9]: Please also copy `package-lock.json` also to keep track of the exact version of every package that is installed.
Also use wild card to reduce the number of layers.

File: Todo-List-App/Dockerfile.prod [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+RUN npm run build
+
+# selecting base image for nginx
+FROM nginx:alpine
+
+#copying the static files from builder
+COPY --from=builder app/build /usr/share/nginx/html
+
+#It exposes the port no, in which the project will be run with docker container. nginx will listen to this port.
+EXPOSE 3000
```
Comment (musfique-m) [inline Todo-List-App/Dockerfile.prod:28]: Why are port 3000 exposed here?
nginx listens to port 80 by default. This EXPOSE instruction should indicate what ports are we gonna get our output from this Dockerfile.

File: Todo-List-App/docker-composer.prod.yml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+version: '3.4'
+services:
+  nginx-todo:
+    stdin_open: true
```
Comment (musfique-m) [inline Todo-List-App/docker-composer.prod.yml:4]: Why are we running the container in interactive mode?

File: Todo-List-App/docker-composer.prod.yml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+    build:
+      context: .
+      dockerfile: Dockerfile.prod
+    image: nginx-todo
+    container_name: nginx-todo
+    ports:
+      - 8080:80
+    environment:
+      - CHOKIDAR_USEPOLLING=true
```
Comment (musfique-m) [inline Todo-List-App/docker-composer.prod.yml:13]: Why do we need to do this? We are not going to watch for file changes in production environment.

**Review by musfique-m — CHANGES_REQUESTED** [review]
Please look at the comments.
Also please remove comments from `Dockerfile.prod`.

File: Todo-List-App/Dockerfile.prod (hunk 1 above)
Comment (D-sourav-M044) [reply Todo-List-App/Dockerfile.prod]: Updated Code. [4999d99](https://github.com/CraftsmenLtd/coe-batch-q322/commit/4999d99522c541a8daa56ea9b2ce5b8224cf1280)

File: Todo-List-App/docker-composer.prod.yml (hunk 4 above)
Comment (D-sourav-M044) [reply Todo-List-App/docker-composer.prod.yml]: Removed the environment variable as it is unnecessary. [4999d99](https://github.com/CraftsmenLtd/coe-batch-q322/commit/4999d99522c541a8daa56ea9b2ce5b8224cf1280)

File: Todo-List-App/docker-composer.prod.yml (hunk 3 above)
Comment (D-sourav-M044) [reply Todo-List-App/docker-composer.prod.yml]: Updated code. [4999d99](https://github.com/CraftsmenLtd/coe-batch-q322/commit/4999d99522c541a8daa56ea9b2ce5b8224cf1280)

File: Todo-List-App/Dockerfile.prod (hunk 2 above)
Comment (D-sourav-M044) [reply Todo-List-App/Dockerfile.prod]: Updated EXPOSE instruction. [1184340](https://github.com/CraftsmenLtd/coe-batch-q322/commit/1184340ca37e2a02f41c116ddb2a6c76483702a4)

**Review by musfique-m — APPROVED** [review]
Looks good.

**Pull request merged**